### PR TITLE
reference/configuration: update tikv configuration file

### PR DESCRIPTION
**Pull request by toutdesuite**
### What is changed, added or deleted? (Required)

+ Add shared block cache related configs
+ Add Titan newly added configs

### Which TiDB version(s) do your changes apply to? (Required)

- [x] master (the latest development version)
- [x] v4.0 (TiDB 4.0 versions)
- [ ] v3.1 (TiDB 3.1 versions)
- [ ] v3.0 (TiDB 3.0 versions)
- [ ] v2.1 (TiDB 2.1 versions)

**If you select two or more versions from above**, to trigger the bot to cherry-pick this PR to your desired release version branch(es), you **must** add corresponding labels such as **needs-cherry-pick-4.0**, **needs-cherry-pick-3.1**, **needs-cherry-pick-3.0**, and **needs-cherry-pick-2.1**.

### What is the related PR or file link(s)?

- This PR is translated from:https://github.com/pingcap/docs-cn/pull/2541
- Other reference link(s):

### CODE REVIEW

File: reference/configuration/tikv-server/configuration-file.md [hunk 1]
```diff
@@ -257,6 +257,21 @@ Configuration items related to storage
 + Default value: `2GB`
 + Unite: MB|GB
 
+## storage.block-cache
+
+Configuration items related to block cache and shared by RocksDB column families. When these configuration items are enabled, block cache separately configured for each column family is disabled.
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:262]:
```suggestion
Configuration items related to the sharing of block cache among multiple RocksDB Column Families (CF). When these configuration items are enabled, block cache separately configured for each column family is disabled.
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 2]
```diff
@@ -257,6 +257,21 @@ Configuration items related to storage
...
+
+### `shared`
+
++ Enables or disables the sharing of block cache.
++ Default value: `true`.
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:267]:
```suggestion
+ Default value: `true`
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 3]
```diff
@@ -257,6 +257,21 @@ Configuration items related to storage
...
+
+### `capacity`
+
++ The size of the shared block cache.
++ Default value: 45% of the size of total system memory.
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:272]:
```suggestion
+ Default value: 45% of the size of total system memory
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 4]
```diff
@@ -257,6 +257,21 @@ Configuration items related to storage
...
++ Unite: KB|MB|GB
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:273]:
```suggestion
+ Unit: KB|MB|GB
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 5]
```diff
@@ -1030,6 +1045,25 @@ Configuration items related to `rocksdb.defaultcf.titan`
 + Minimum value: `0`
 + Unit: KB|MB|GB
 
+### `blob-run-mode`
+
++ Specifies the running mode of Titan.
++ Available values:
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:1051]:
```suggestion
+ Optional values:
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 6]
```diff
@@ -1030,6 +1045,25 @@ Configuration items related to `rocksdb.defaultcf.titan`
...
+    + `normal`: Writes data to the blob file when the value size exceeds `min-blob-size`.
+    + `read_only`: Reads the original data in the blob file, but refuses to write new data to the blob file.
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:1053]:
```suggestion
    + `read_only`: Refuses to write new data to the blob file, but still reads the original data from the blob file.
```

File: reference/configuration/tikv-server/configuration-file.md [hunk 7]
```diff
@@ -1030,6 +1045,25 @@ Configuration items related to `rocksdb.defaultcf.titan`
...
++ Default value: `normal`
+
+### `level-merge`
+
++ Determines whether to optimize the read performance. When `level-merge` is enabled, there is more write amplification.
++ Default value: `true`
+
+### `gc-merge-rewrite`
+
++ Determines whether to write back Titan GC using the merge operator. When `gc-merge-rewrite` is enabled, it reduces the effect of Titan GC on the writes in the foreground.
```
Comment (TomShawn) [inline reference/configuration/tikv-server/configuration-file.md:1064]: The original Chinese `是否开启使用 merge operator 来进行 Titan GC 写回操作，减少 Titan GC 对于前台写入的影响。` is not clear. @Connor1996 PTAL

File: reference/configuration/tikv-server/configuration-file.md [hunk 8]
```diff
@@ -1030,6 +1045,25 @@ Configuration items related to `rocksdb.defaultcf.titan`
...
```
Comment (Connor1996) [inline reference/configuration/tikv-server/configuration-file.md:1064]:
```suggestion
+ Determines whether to use the merge operator to write back blob indexes for Titan GC. When `gc-merge-rewrite` is enabled, it reduces the effect of Titan GC on the writes in the foreground.
```

**Comment by toutdesuite** [comment]
@TomShawn PTAL. Thanks!

**Comment by TomShawn** [comment]
LGTM
@Connor1996 PTAL again

**Pull request merged**